geo/src/algorithm/rotate.rs

```diff
@@ -62,60 +40,31 @@ pub trait Rotate<T> {
     ///
     /// assert_relative_eq!(expected, rotated);
     /// ```
-    fn rotate_around_centroid(&self, angle: T) -> Self
-    where
-        T: CoordFloat;
+    #[must_use]
+    fn rotate_around_centroid(&self, degrees: T) -> Self;
+
+    /// Mutable version of [`Self::rotate_around_centroid`]
+    fn rotate_around_centroid_mut(&mut self, degrees: T);
 
     /// Rotate a geometry around the center of its [bounding box](BoundingRect) by an angle, in
     /// degrees.
     ///
     /// Positive angles are counter-clockwise, and negative angles are clockwise rotations.
     ///
-    /// ## Performance
-    ///
-    /// If you will be performing multiple transformations, like [`Scale`](crate::Scale),
-    /// [`Skew`](crate::Skew), [`Translate`](crate::Translate), or [`Rotate`](crate::Rotate), it is more
-    /// efficient to compose the transformations and apply them as a single operation using the
-    /// [`AffineOps`](crate::AffineOps) trait.
-    ///
-    /// # Units
-    ///
-    /// - `angle`: degrees
-    fn rotate_around_center(&self, angle: T) -> Self
-    where
-        T: CoordFloat;
-
-    #[deprecated(
-        note = "Equivalent to `rotate_around_centroid` except for `Polygon<T>`,
-                    where it is equivalent to rotating around the polygon's outer ring.
-                    Call that instead, or `rotate_around_center` if you'd like to rotate
-                    around the geometry's bounding box center."
-    )]
-    fn rotate(&self, angle: T) -> Self
```

Comment:
Note I'm removing this deprecated method. The deprecation was merged about a year ago in a3470cd3f5b1b658e5ba29c057f947445635095d, but was released about 4 months ago in geo-0.19.0 (releases are more frequent lately 😄).